### PR TITLE
growth_constraints.md

### PR DESCRIPTION
**Pull request by LaplacetteCM**
This is the proposed wiki for growth_constraints.py

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/docs/growth_constraints.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+The `GrowthConstraints` class handles the environmental constraints to potential plant growth that a crop
+face during the growing season. This module addresses the stresses caused by insufficient
```
Comment (morrowcj) [inline SC_redesign/Crop_and_Soil/docs/growth_constraints.md]:
```suggestion
faces during the growing season. This module addresses the stresses caused by insufficient 
```

File: SC_redesign/Crop_and_Soil/docs/growth_constraints.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+water, inadequate nutrients (nitrogen and phosphorus) and extreme temperatures, and determines which, if any, 
+is limiting plant growth on a given day.  
+
+The main method, `constrain_growth()` calls on the **water stress**, **temperature stress**, **nitrogen 
+stress** and **phosphorus stress** methods to provide daily updates on the **plant growth factor**, 
+which quantifies the fraction of potential plant growth achieved on a given day. Each environmental stress 
+varies from 0, under optimal conditions, to 1, under extreme stress. 
+
+**Water stress** calculates the stress caused by the water soil conditions on a given day by comparing 
+actual and potential plant transpiration according to:
```
Comment (morrowcj) [inline SC_redesign/Crop_and_Soil/docs/growth_constraints.md]:
```suggestion
**Water stress** is the stress caused by the water soil conditions on a given day by comparing 
actual and potential plant transpiration:
```

File: SC_redesign/Crop_and_Soil/docs/growth_constraints.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+     1- exp\frac{-0.1054\times (T_{\text{crop, opt}} - T_{\text{air}})^{2}}{((2\times T_{\text{crop, opt}} - T_{\text{crop, min}}) - T_{\text{crop, min}})^{2}}        & \text{if } T_{\text{crop, opt}} < T_{\text{air}} <= 2\times T_{\text{crop, opt}} - T_{\text{crop, min}} \\                                                                                                                                                                                                                                                   
+     1                                                                                                                                                                 & \text{otherwise}
+    
+  \end{cases}
+$$
+```
+where $T_{\text{stress}}$ is `temp_stress`, $T_{\text{air}}$ is `temperature`, the mean air
+temperature for a given day, $T_{\text{crop, min}}$ is `minimum_temperature`, the base or minimum temperature for the crop 
+to grow, and $T_{\text{opt}}$ is `optimal_temperature`, the optimal temperature for the crop to grow.
```
Comment (morrowcj) [inline SC_redesign/Crop_and_Soil/docs/growth_constraints.md]: One alternative way to write these things may be a bit simpler:

```suggestion
where $T_{\text{stress}}$ is the temperature stress (`temp_stress`), $T_{\text{air}}$ is the mean air
temperature (`temperature`) for a given day, $T_{\text{crop, min}}$ is the base or minimum temperature for the crop 
to grow (`minimum_temperature`), and $T_{\text{opt}}$ is the optimal temperature for the crop to grow (`optimal_temperature`). 
```

**Review by morrowcj — APPROVED** [review]
This looks excellent.  I made a few minor comments about typos and format. Well done, and thanks again!

File: SC_redesign/Crop_and_Soil/docs/growth_constraints.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+The main method, `constrain_growth()` calls on the **water stress**, **temperature stress**, **nitrogen 
+stress** and **phosphorus stress** methods to provide daily updates on the **plant growth factor**, 
+which quantifies the fraction of potential plant growth achieved on a given day. Each environmental stress 
+varies from 0, under optimal conditions, to 1, under extreme stress. 
+
+**Water stress** calculates the stress caused by the water soil conditions on a given day by comparing 
+actual and potential plant transpiration according to:
+
+```math
```
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/docs/growth_constraints.md]: So in previews everything inside the ` ```math` notation shows up with the `$$` symbols, in both VS Code's preview feather, and in a markdown preview tool online. 

Seems like you only need one notation or the other (e.g. ```math or a $$): see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions#writing-expressions-as-blocks

**Review by JoeWaddell — CHANGES_REQUESTED** [review]
The writing looks good (though I can't comment much on the actual science behind it - but I trust you two!). Just note the comment about math formatting - it might just be my display settings, but the preview shows funky formatting because of using the $$ notation nested inside the math block. Once that's fixed it should be clean! Nice work

File: SC_redesign/Crop_and_Soil/docs/growth_constraints.md (hunk 4 above)
Comment (JoeWaddell) [reply SC_redesign/Crop_and_Soil/docs/growth_constraints.md]: So you can see what I mean: 
image


Removing the math block: 
image

Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/docs/growth_constraints.md]: I didn't know VS code could handle LaTeX math, Pycharm cannot natively, unfortunately.

Comment (LaplacetteCM) [reply SC_redesign/Crop_and_Soil/docs/growth_constraints.md]: Thanks Joe! I've updated the file deleting the math block and worked. I'll update the rest of the wikis the same way.

**Review by JoeWaddell — APPROVED** [review]
That did the trick: looks nice!

**Pull request merged**